### ISSUE TITLE
Remove all maxWidth props from the codebase

### ISSUE THREAD

**Issue by lucasbordeau** [twentyhq/core-team-issues#163]
## Scope & Context

The `maxWidth` and other `width` props are a workaround for lack of a better dynamic solution. It should be removed entirely from the codebase.

The only places where we set width should be on the most generic containers like dropdown or right drawer, everything else should derive its width from the parent container.

**Comment by gvkhna** [comment]
I encountered this limitation here, but hardcoded a value to get a POC.

https://github.com/twentyhq/twenty/pull/6943

There shouldn't be any maxWidth I agree, the entire layout should flex to adjust. A "Double Word" UI Test would be helpful to see how much would be truncated typically for Germans etc.

**Comment by lucasbordeau** [comment]
@Bonapara Could we define an enum for widths ? For example, could dropdown widths be reduced to 2 or 3 use cases : `NARROW`, `MEDIUM`, `LARGE` ? And define the default one for a maximum of components ?

Then we could expose this prop only for components that must know it, to avoid wrong UX behavior.

This way we could improve our DX a lot and avoid hard-coded widths and regression.

A lot of component could just be refactored without being passed any width prop, but some will surely need this enum because it's more a UI choice than a dev refactor.

**Comment by Bonapara** [comment]
Narrow: 160px  
Medium: 200px  
Large: 240px  
ExtraLarge: 320px

**Comment by lucasbordeau** [comment]
This issue should focus on 2 things : 
- Abstracting the dropdown possible widths with an enum given by Bonapara
- Make sure we only set the width once on the highest parent component

**Comment by lucasbordeau** [comment]
@charlesBochet I'm going to add more details about this issue so we can proceed incrementally.

**Comment by charlesBochet** [comment]
I disagree with: "Make sure we only set the width once on the highest parent component"

Rational, let's discuss it tomorrow

<img width="1049" alt="Image" src="https://github.com/user-attachments/assets/2fe455c3-598d-4aaf-9652-bcee28be8a19" />
(sorry it's french)

**Comment by lucasbordeau** [comment]
Ok I agree, I should have said that "we should make sure that we only set the width only once in the most relevant parent component", but of course since for the same dropdown, the content width can vary, then the top most component that should be responsible for the width seems to be the content container, or in the case of this filter dropdown, the different containers that can be displayed as children.

The problem is that we cannot so easily find a generic solution because there is logic with middleware to handle auto-resizing of dropdown instances. Let's think about it.

**Comment by lucasbordeau** [comment]
I'm closing this issue because the main problem we have is about dropdown content width, which will be tackled in another issue : https://github.com/twentyhq/core-team-issues/issues/951
